sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosBulkItemResponse.java

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.cosmos.implementation.JsonSerializable;
+import com.azure.cosmos.implementation.batch.BatchExecUtils;
+import com.azure.cosmos.util.Beta;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import reactor.core.publisher.Flux;
+
+import java.time.Duration;
+import java.util.Map;
+
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
+
+/**
+ * Response of a {@link CosmosItemOperation} request when processed using Bulk by calling
+ * {@link CosmosAsyncContainer#processBulkOperations(Flux, BulkProcessingOptions)}.
+ *
+ */
+@Beta(Beta.SinceVersion.V4_8_0)
+public final class CosmosBulkItemResponse {
+
+    private final String eTag;
+    private final double requestCharge;
+    private final ObjectNode resourceObject;
+    private final int statusCode;
+    private final Duration retryAfter;
+    private final int subStatusCode;
+    private final Map<String, String> responseHeaders;
+    private final CosmosDiagnostics cosmosDiagnostics;
+
+    /**
+     * Initializes a new instance of the {@link CosmosBulkItemResponse} class.
+     */
+    CosmosBulkItemResponse(String eTag,
+                           double requestCharge,
+                           ObjectNode resourceObject,
+                           int statusCode,
+                           Duration retryAfter,
+                           int subStatusCode,
+                           Map<String, String> responseHeaders,
+                           CosmosDiagnostics cosmosDiagnostics) {
+
+        checkNotNull(statusCode, "expected non-null statusCode");
```

Comment:
nit: status code is int, so may never be null?

Reply:
Yes, backend always returns a status code, fi there is no exception. The same is true for CosmosItemResponse.

Reply:
Agreed. Will remove the check. Thanks.